alf/experience_replayers/replay_buffer.py

```diff
@@ -32,21 +34,216 @@ def __init__(self,
                  max_length=1024,
                  device="cpu",
                  allow_multiprocess=False,
+                 her_k=0,
+                 step_type_field="step_type",
+                 achieved_goal_field="observation.achieved_goal",
+                 desired_goal_field="observation.desired_goal",
```

Comment:
I also don't think it's a good idea to add many seemly irrelevant arguments like "achieved_goal_field" in a general class like ReplayBuffer. The current implementation is overly bloated.